### ISSUE TITLE
Incompatible with GNOME 43

### ISSUE THREAD

**Issue by MANICX100** [mymindstorm/gnome-volume-control#31]
Tested in Fedora 37 beta and forced it to run with metadata.json edit.

Does not work

**Comment by Aasikki** [comment]
Makes sense because the whole menu is redesigned in gnome 43 and has new audio options. Hopefully this can be updated soon enough as sadly the new audio options don't include this feature.

**Comment by dmzoneill** [comment]
+1

**Comment by Webxorcist** [comment]
Is this project still active?

**Comment by mymindstorm** [comment]
I’m too busy to maintain at the moment.

On Oct 31, 2022, at 10:56 AM, Webxorcist ***@***.***> wrote:

﻿

Is this project still active?

**Comment by obfuscated-loop** [comment]
Shame, this is a really good extension.

This does seem possible to do though - the main change with GNOME 43 is that it *"removed aggregateMenu and uses quick settings (quickSettings) instead"*.

I'm playing around with it now in a GNOME sandbox, I've never worked with GJS before so I'm just trying to figure it out with the debugger. `ApplicationStreamSlider` should now extend `imports.ui.quickSettings.QuickSlider` and `Main.panel.statusArea.aggregateMenu._volume.menu.addMenuItem(volumeMixer);` should be something like `Main.panel.statusArea.quickSettings._addItems(volumeMixer, 2);`.. I think?

Again, never worked with extensions before. Guess I'll have to try to learn.

Sources: https://gjs.guide/extensions/upgrading/gnome-shell-43.html#quick-settings and https://gjs.guide/extensions/topics/quick-settings.html#slider

**Comment by mymindstorm** [comment]
@james-still you are more than welcome to submit a PR. If I don't respond to a PR within a week or two, just ping me again.

**Comment by onlyreportingissues** [comment]
> Shame, this is a really good extension.
> 
> This does seem possible to do though - the main change with GNOME 43 is that it _"removed aggregateMenu and uses quick settings (quickSettings) instead"_.
> 
> I'm playing around with it now in a GNOME sandbox, I've never worked with GJS before so I'm just trying to figure it out with the debugger. `ApplicationStreamSlider` should now extend `imports.ui.quickSettings.QuickSlider` and `Main.panel.statusArea.aggregateMenu._volume.menu.addMenuItem(volumeMixer);` should be something like `Main.panel.statusArea.quickSettings._addItems(volumeMixer, 2);`.. I think?
> 
> Again, never worked with extensions before. Guess I'll have to try to learn.
> 
> Sources: https://gjs.guide/extensions/upgrading/gnome-shell-43.html#quick-settings and https://gjs.guide/extensions/topics/quick-settings.html#slider

Maybe this can help you: https://github.com/qwreey75/quick-settings-tweaks

**Comment by Aasikki** [comment]
> > Shame, this is a really good extension.
> > This does seem possible to do though - the main change with GNOME 43 is that it _"removed aggregateMenu and uses quick settings (quickSettings) instead"_.
> > I'm playing around with it now in a GNOME sandbox, I've never worked with GJS before so I'm just trying to figure it out with the debugger. `ApplicationStreamSlider` should now extend `imports.ui.quickSettings.QuickSlider` and `Main.panel.statusArea.aggregateMenu._volume.menu.addMenuItem(volumeMixer);` should be something like `Main.panel.statusArea.quickSettings._addItems(volumeMixer, 2);`.. I think?
> > Again, never worked with extensions before. Guess I'll have to try to learn.
> > Sources: https://gjs.guide/extensions/upgrading/gnome-shell-43.html#quick-settings and https://gjs.guide/extensions/topics/quick-settings.html#slider
> 
> Maybe this can help you: https://github.com/qwreey75/quick-settings-tweaks

Dang this is awesome, thanks!

**Comment by jfernandz** [comment]
I'm wondering if some of you have a fork where is developing, but it seems to be you are on the right way :)

**Comment by Caffe1neAdd1ct** [comment]
Started looking at this last night.

Would it be better to use a separate menu or integrate with the main quick settings dropdown?

- https://gjs.guide/extensions/topics/quick-settings.html#example-usage

With the quick settings a ToggleMenu doesn't really make a lot of sense as we don't need the toggle aspect, what would be better to use here to provide the popup menu containing all the sliders?

![image](https://user-images.githubusercontent.com/1327332/210537077-6559b8a3-f405-4bdd-8c3e-db6c82003ab6.png)

**Comment by TimeLincoln** [comment]
can you extend the sound output device chooser menu that already exists?

**Comment by Martinligabue** [comment]
forcing the extension now throws the following error: `class heritage Volume$1.StreamSlider is not an object or null`
